cscs-checks/libraries/gridtools/gridtools_tests.py

```diff
@@ -0,0 +1,203 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.parameterized_test(['vertical_advection_dycore_naive'],
+                        ['vertical_advection_dycore_mc'],
+                        ['simple_hori_diff_naive'], ['simple_hori_diff_mc'],
+                        ['vertical_advection_dycore_cuda'],
+                        ['simple_hori_diff_cuda'])
+class GridToolsCheck(rfm.RegressionTest):
+    def __init__(self, variant):
+        super().__init__()
+
+        # Check if this is a device check
+        is_cuda_test = "cuda" in variant
+
+        self.descr = 'GridTools test base'
+
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['CMake', 'Boost']
+        if self.current_system.name is 'daint':
+            self.modules.append('gcc/5.3.0')
+        else:
+            self.modules.append('gcc')
```

Comment:
For me I don't need to load gcc for dom, it is already loaded by `PrgEnv-gnu`. Only if you wanted a particular version it would make sense to append it.

Reply:
Also the gcc/5.3 will fail as soon as we upgrade Daint and we will have to change the test. The test should target Dom only, but Daint should still be on its valid systems so as to be ready.

Reply:
Yes sorry, this is useless.

Reply:
@vkarak what should I do for Daint+gcc 5.3.0? What if we leave that part as it is and then we delete the gcc/5.3.0 loading command as soon as that version will not be available anymore on Daint? Otherwise, is it ok if I remove Daint from the valid system list and I leave the reference numbers?

Reply:
You remove `gcc/5.3.0` completely and you still keep Daint in valid systems. The tests in master target CLE7, so we don't care if it doesn't run on current Daint.

Reply:
Ok, done.